<launch>
	<param name="robot_description" command="cat $(find cepheus_robot)/model/cepheus.urdf" />   

	<rosparam command="load" file="$(find cepheus_robot)/config/controllers.yaml"/>

	<node name="cepheus_interface_node" pkg="cepheus_robot" type="cepheus_interface_node" respawn="false" output="screen" required="true" >
		<param name="thruster_force" value="0.6"/>
		<param name="max_motor_current" value="2.00"/>
		<param name="rw_max_speed" value="400"/>
		<param name="rw_total_inertia" value="0.00197265"/>
		<param name="rw_max_torque" value="0.02"/>
		<param name="rw_max_power" value="30"/>
		<param name="loop_rate" value="200.0"/>
		
		<!--param name="left_shoulder_limit_pos" value="3.1548" /-->
		<param name="left_shoulder_limit_pos" value="1.34" />

		<!--param name="left_elbow_limit_pos" value="1.658" /-->
		<param name="left_elbow_limit_pos" value="2.22" />
		<param name="right_shoulder_limit_pos" value="-2.453" />
		<!--param name="right_elbow_limit_pos" value="-1.6231" /-->
		<param name="right_elbow_limit_pos" value="1.811219" />
		<!-- For the param below : False if no planner needed. It exists for gripper feedback to the planner. Nessesary if the robots tracks, chases and tries to grasp something -->
		<param name="use_with_chase_planner" value="True" />          
	</node>

	<node name="cepheus_ctrl" pkg="cepheus_robot" type="cepheus_ctrl" respawn="false" output="screen" required="true">
		<param name="controllers_to_spawn" value="joint_state_publisher	reaction_wheel_effort_controller"/>
	</node>
	
	<node pkg="rosserial_python" type="serial_node.py" name="serial_node" output="screen">
		<param name="port" value="/dev/ttyACM0"/>
		<param name="baud" value="115200"/>
	</node>

	<node name="cepheus_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" output="screen" required="true">
		<param name="publish_frequency" value="200"/>
	</node>

	<node name="alex_gripper_control_node" pkg="cepheus_robot" type="alex_gripper_control_node" output="screen" required="true">
		<param name="loop_rate" value="200"/>
	</node>

	<node name="record_topics_for_verification" pkg="rosbag" type="record" args="-o $(find cepheus_robot)/bags/cepheus /secs /set_left_shoulder_effort /set_left_elbow_effort /set_right_elbow_effort /set_left_shoulder_qd /set_left_elbow_qd /set_right_elbow_qd /read_left_shoulder_position /read_left_elbow_position /read_right_elbow_position /read_left_shoulder_velocity /read_left_elbow_velocity /read_right_elbow_velocity" respawn="true" output="screen"/>

	<!--node name="base_controller" pkg="cepheus_robot" type="base_controller_node" output="screen" required="true">
	<param name="use_reaction_wheel" value="True"/>
			<param name="loop_rate" value="200.0"/>
			<param name="use_with_planner" value="True"/>
			<param name="kp_gain_lin" value="1.75"/>
			//<param name="kp_gain_lin" value="0.3"/>
			<param name="kp_gain_ang" value="1.0"/>
			<param name="kd_gain_lin" value="12.0"/>
			//<param name="kd_gain_lin" value="3.0"/>
			<param name="kd_gain_ang" value="1.24"/>
	</node-->

	<!--node name="base_planner" pkg="cepheus_robot" type="base_planner_node" output="screen">
		<param name="time_step" value="0.0100"/>
		<param name="loop_rate" value="200.0"/>
		<param name="target_speed" value="0.02"/>
		<param name="target_accel" value="0.005"/>
		<param name="probe_offset" value="0.523"/>
	</node-->  

	<!--node name="new_base_planner" pkg="cepheus_robot" type="new_base_planner_node" output="screen" required="true">
	</node-->

	<!--include file="$(find phase_space)/launch/track_two_objects.launch">
		<arg name="turn_rviz_on" value="false" />
		<arg name="turn_phase_space_on" value="true" />
	</include-->

	<!--node pkg="tf2_ros" type="static_transform_publisher" name="gripper_target_broadcaster" args="0.215 0.215 0 0 0 0 assist_robot right_gripper_target" /-->
	<!--node pkg="tf2_ros" type="static_transform_publisher" name="left_hand_base_broadcaster" args="0.172628 0.091404 0 0 0 0 cepheus left_hand_base" /-->
	<!--node pkg="tf2_ros" type="static_transform_publisher" name="right_hand_base_broadcaster" args="0.172628 -0.091404 0 0 0 0 cepheus right_hand_base" /-->

</launch>
